compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -757,30 +763,42 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
         canConstrain(param2) && canInstantiate(param2) ||
         compareLower(bounds(param2), tyconIsTypeRef = false)
       case tycon2: TypeRef =>
-        isMatchingApply(tp1) ||
-        compareLower(tycon2.info.bounds, tyconIsTypeRef = true)
+        isMatchingApply(tp1) || {
+          tycon2.info match {
+            case info2: TypeBounds =>
```

Comment:
Complication: Need to compare baseclasses of lhs against full rhs if it is an AppliedType